bin/sepolia-bundle-lander/src/intent_builder.rs

```diff
@@ -333,4 +338,26 @@ where
             % 2
             == 0
     }
+
+    async fn get_valid_nonce(&self, user: Address) -> eyre::Result<u64> {
+        let nonce_tracker = Nonces::new(self.angstrom_address);
+
+        let mut nonce: u64 = rand::random();
+        loop {
+            let slot = nonce_tracker.get_nonce_word_slot(user, nonce);
+
+            let word = self
+                .provider
+                .get_storage_at(self.angstrom_address, slot.into())
+                .await?;
+
+            let flag = U256::from(1) << (nonce as u8);
+
+            if (word ^ flag) & flag == flag {
+                break Ok(nonce);
+            } else {
+                nonce = rand::random();
+            }
+        }
+    }
```

Comment:
just use the agstrom api to get a valid nonce.
`let nonce = self.client.valid_nonce(user).await.unwrap();`